src/rez/plugin_managers.py

```diff
@@ -52,18 +53,23 @@ def extend_path(path, name):
     init_py = "__init__" + os.extsep + "py"
     path = path[:]
 
-    for dir_ in config.plugin_path:
-        if not os.path.isdir(dir_):
-            if config.debug("plugins"):
-                print_debug("skipped nonexistant rez plugin path: %s" % dir_)
-            continue
+    def append_if_valid(dir_):
+        if os.path.isdir(dir_):
+            subdir = os.path.normcase(os.path.join(dir_, pname))
+            initfile = os.path.join(subdir, init_py)
+            if subdir not in path and os.path.isfile(initfile):
+                path.append(subdir)
 
-        subdir = os.path.join(dir_, pname)
-        # XXX This may still add duplicate entries to path on
-        # case-insensitive filesystems
-        initfile = os.path.join(subdir, init_py)
-        if subdir not in path and os.path.isfile(initfile):
-            path.append(subdir)
+        elif config.debug("plugins"):
+            print_debug("skipped nonexistant rez plugin path: %s" % dir_)
+
+    # Extend old-style plugins
+    for dir_ in config.plugin_path:
+        append_if_valid(dir_)
+    # Extend new-style plugins
+    for loader, name, ispkg in pkgutil.iter_modules():
```

Comment:
I don't quite follow why this is here? Do new-style plugins require an `extend_path`, as old-style do?

In any case I don't think we can do this. This is causing an iteration over all modules, per plugin. I'm already a little concerned about doing that once, let alone once per plugin load. We need to come up with a different approach that doesn't do this here - module iteration should happen once only. Also imo new-style plugins shouldn't require use of `extend_path` at all.